checkers/invalid_range_index_checker.py

```diff
@@ -0,0 +1,49 @@
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from ast import literal_eval
+
+
+class InvalidRangeIndexChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = 'custom'
+    msgs = {'E9993':
+                ('You should not use invalid range index on line %s',
+                 'invalid-range-index',
+                 'Used when you use invalid index range')}
+    options = ()
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("invalid-range-index")
+    def visit_call(self, node):
+        if isinstance(node.func, astroid.Name):
+            name = node.func.name
+            # ignore the name if it's not a builtin (i.e. not defined in the
+            # locals nor globals scope)
+            if not (name in node.frame() or name in node.root()) and name == 'range':
+                # the arguments of 'range' call
+                arg = node.args
+                lst = list(map(lambda z: literal_eval(z.as_string()), arg))
+                # check if there is no args in 'range' call
+                if len(arg) == 0 or \
```

Comment:
Change the formatting of the compound condition as follows:

```python
if (len(arg) == 0 or
    not all(...) or
    (len(arg) == 1 and lst[0] < 2) or
    ...):
```

Each condition separated by an `or` is on a new line, vertically aligned, with all the `or`s going at the ends of the lines.